WinImageBuilder.psm1

```diff
@@ -553,6 +553,34 @@ function Download-CloudbaseInit {
     }
 }
 
+function Download-QemuGuestAgent {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$QemuGuestAgentConfig,
+        [Parameter(Mandatory=$true)]
+        [string]$ResourcesDir,
+        [Parameter(Mandatory=$true)]
+        [string]$OsArch
+    )
+
+    $QemuGuestAgentUrl = $QemuGuestAgentConfig
+    if ($QemuGuestAgentConfig -eq 'True') {
+        $arch = "x86"
+        if ($OsArch -eq "AMD64") {
+            $arch = "x64"
+        }
+        $QemuGuestAgentUrl = "https://fedorapeople.org/groups/virt/virtio-win/direct-downloads" + `

```

Comment:
To be honest, I wasn't aware that a separate download link actually exists just for the agent :). I always installed the agent from the VirtIO driver ISO.

However, I would make the base URL for the agent download, configurable. There are 3 main reasons for that:

- The upstream URL may change, and that will mean changing the code to get it working again
- Certified VirtIO drivers come from different vendors, and with them, probably a bundled version of the guest agents.
- There may be limited network connectivity in the environment where this may run, and access to ```fedorapeople.org``` may not be available.

Reply:
see config.psm1 description:

**The value can be changed to a custom URL, to allow other QEMU guest agent versions to be installed.**

Reply:
Ohh, right. I thought that was a boolean value, but you can shadow it to include an URL instead of a boolean. I would recommend not doing that, as it's highly confusing, and can lead to really awkward code paths. Instead I would recommend a separate config value where you can simply specify the installer URL or a base URL.

But I leave this decision up to you.

Reply:
I should change the name of the config value, to something more descriptive. Like install_qemu_guest_agent=False|True.
And another config value, qemu_guest_agent_source:None (same behaviour) as present one, from_virtio_drivers_path or a custom URL.

Currently, I am leaving this PR open, to see what are the wanted use cases from the community.